src/dcast.d

```diff
@@ -677,12 +677,19 @@ extern (C++) MATCH implicitConvTo(Expression e, Type t)
                 }
                 else
                 {
+                    if (e.basis)
+                    {
+                        MATCH m = e.basis.implicitConvTo(telement);
+                        if (m < result)
+                            result = m;
+                    }
                     for (size_t i = 0; i < e.elements.dim; i++)
                     {
                         Expression el = (*e.elements)[i];
                         if (result == MATCHnomatch)
                             break;
-                        // no need to check for worse
+                        if (!el)
```

Comment:
Yes, but this PR leaves them null and lazily fills them w/ `basis` thereby avoid multiple semantic calls on the same element.